### PR TITLE
Add vscode keybindings

### PR DESCRIPTION
**Pull request by benhoman**
This PR adds a `userKeybindings` option that allows a user to create custom keybindings in the nix configuration. This option was mostly copied from the existing `userSettings` option.

### CODE REVIEW

File: modules/programs/vscode.nix [hunk 1]
```diff
@@ -28,6 +28,27 @@ in
         '';
       };
 
+      userKeybindings = mkOption {
+        type = types.listOf types.attrs;
```
Comment (rycee) [inline modules/programs/vscode.nix]: Would it be possible to have this with `types.attrsOf types.str`. I.e., the example becomes
```
{
  "ctrl+h" = "workbench.action.navigateLeft";
  "ctrl+l" = "workbench.action.navigateRight";
}
```

Comment (wedens) [reply modules/programs/vscode.nix]: It's possible to have the same keybindings with different `when`. For example (from the doc):
```json
{ "key": "f5",              "command": "workbench.action.debug.continue", "when": "inDebugMode" },
{ "key": "f5",              "command": "workbench.action.debug.start",    "when": "!inDebugMode" },
```

**Comment by fabianpage** [comment]
Hi

What's the state of this PR, does't it need some more work? (I'm a nix beginner but maybe i could help)

**Comment by Rizary** [comment]
it's duplicate by https://github.com/rycee/home-manager/pull/1351 and it seems the #1351 is more close to what vscode want?

**Comment by rycee** [comment]
I'll close this since I believe this functionality is available in master now.

**Pull request closed without merging**